### PR TITLE
Fix the bug when the input_size of the model was not defined

### PR DESCRIPTION
**Pull request by twolodzko**
We were saving metrics on `on_train_begin` but there is an edge case in TensorFlow where the model is not ready yet before the training starts, so our integration was crashing. The test was adjusted to catch that (it fails correctly).

### CODE REVIEW

File: tests/test_e2e.py [hunk 1]
```diff
@@ -28,6 +32,19 @@ def test_e2e(dataset, model, log_model_diagram, log_on_batch):
         validation_data=(x_test, y_test),
     )
 
+    # retry if Neptune didn't reresh its cache
```
Comment (kshitij12345) [inline tests/test_e2e.py:35]: `reresh` to `refresh`

File: neptune_tensorflow_keras/impl/__init__.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def _log_metrics(self, logs, category: str, trigger: str):
             except NeptuneException:
                 pass
 
-    def on_train_begin(self, logs=None):  # pylint:disable=unused-argument
+    def on_train_end(self, logs=None):  # pylint:disable=unused-argument
```
[the remark this answers is not in the document]
Comment (twolodzko) [reply neptune_tensorflow_keras/impl/__init__.py]: Fixed.

File: tests/test_e2e.py (hunk 1 above)
Comment (twolodzko) [reply tests/test_e2e.py]: Thanks.

File: neptune_tensorflow_keras/impl/__init__.py [hunk 3]
```diff
@@ -137,19 +137,23 @@ def _log_metrics(self, logs, category: str, trigger: str):
                 pass
 
     def on_train_begin(self, logs=None):  # pylint:disable=unused-argument
-        self._model_logger["summary"] = _model_summary_file(self.model)
         self._model_logger["optimizer_config"] = self.model.optimizer.get_config()  # it is a dict
         self._metric_logger["fit_params"] = self.params
 
         if self._log_model_diagram:
```
Comment (kshitij12345) [inline neptune_tensorflow_keras/impl/__init__.py:143]: I think we should also move `_log_model_diagram` to `on_train_end` (I think this might fail as well 🤔 )

**Review by kshitij12345 — APPROVED** [review]
LGTM (given that the CI is green)!

Thanks @twolodzko

File: neptune_tensorflow_keras/impl/__init__.py (hunk 3 above)
Comment (twolodzko) [reply neptune_tensorflow_keras/impl/__init__.py]: You are correct, fixed.
